sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/internal/SensorContextTester.java

```diff
@@ -212,32 +212,39 @@ public NewIssue newIssue() {
 
   @CheckForNull
   public Integer lineHits(String fileKey, int line) {
-    Optional<Integer> sum = sensorStorage.coverageByComponent.get(fileKey).stream()
+    return sensorStorage.coverageByComponent.get(fileKey).stream()
       .map(c -> c.hitsByLine().get(line))
-      .flatMap(i -> Stream.of(Optional.ofNullable(i)))
-      .reduce(Optional.empty(),
-        (o1, o2) -> o1.isPresent() ? o1.map(i -> i + o2.orElse(0)) : o2);
-    return sum.orElse(null);
+      .flatMap(Stream::of)
+      .filter(Objects::nonNull)
+      .reduce(null, SensorContextTester::sumOrNull);
+  }
+
+  @CheckForNull
+  public static Integer sumOrNull(Integer o1, Integer o2) {
+    return o1 == null ? o2 : (o1 + o2);
```

Comment:
This only works when we know that both operands will never be `null` at the same time. We know this when calling in `.filter(Objects::nonNull).reduce(null, ...)`, but not here. This hidden assumption could be misleading. It would be better to make this method take a `Stream<Integer>`, and apply the filter and reduce here. Then there will no hidden assumptions, and the method will be safe to use.